printing typealias with array.map causes c error: character array initializer must be a literal, optionally enclosed in braces

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 5c65e58.bc7a844
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 24 cpus, 64bit, little endian, 

getwd: D:\vbug2
vexe: C:\v\v.exe
vexe mtime: 2024-12-09 22:04:50

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\info\.vmodules
VTMP: OK, value: C:\Users\info\AppData\Local\Temp\v_0

Git version: git version 2.42.0.windows.2
Git vroot status: weekly.2024.49-50-gbc7a844a-dirty
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


emcc version: Error: 'emcc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc status: thirdparty-windows-amd64 b425ac82
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg main.v`

```v
module main

type Mat4 = [16]f32

struct GameObject  {
mut:
	transform Mat4
	children []&GameObject
}

fn main() {
	mut v1 := &GameObject{}
	eprintln("children: ${v1.children.map(it.transform)}")
}
```

**What did you expect to see?**

no c error

**What did you see instead?**

```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JEPQN7KB2SC68GBN09Z7E5QJ.tmp.c:3988: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JEPQN7KB2SC68GBN09Z7E5QJ.tmp.c:5975: error: character array initializer must be a literal, optionally enclosed in braces
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzU3NmI4MzFlMDYzMmRkMDhlMjk0MGEiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.pumRQJp9f_a6Ls2jEGS5WZ0CFYce2jYn3-OE_L6tcuk">Huly&reg;: <b>V_0.6-21553</b></a></sub>

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -576,6 +576,28 @@ fn (mut g Gen) gen_array_map(node ast.CallExpr) {
 			g.write('${ret_elem_styp} ${tmp_map_expr_result_name} = ')
 			g.expr(expr.expr)
 		}
+		ast.SelectorExpr {
+			if expr.typ != ast.void_type && g.table.final_sym(expr.typ).kind == .array_fixed {
+				atype := g.styp(expr.typ)
+				if closure_var_decl != '' {
+					g.write('memcpy(&${closure_var_decl}, &')
+					g.expr(expr)
+					g.write(', sizeof(${atype}))')
+				} else {
+					g.writeln('${ret_elem_styp} ${tmp_map_expr_result_name};')
+					g.write('memcpy(&${tmp_map_expr_result_name}, &')
+					g.expr(expr)
+					g.write(', sizeof(${atype}))')
+				}
+			} else {
+				if closure_var_decl != '' {
+					g.write('${closure_var_decl} = ')
+				} else {
+					g.write('${ret_elem_styp} ${tmp_map_expr_result_name} = ')
+				}
+				g.expr(expr)
+			}
+		}
 		else {
 			if closure_var_decl != '' {
 				g.write('${closure_var_decl} = ')

--- a/vlib/v/tests/builtin_arrays/array_map_to_fixed_array_test.v
+++ b/vlib/v/tests/builtin_arrays/array_map_to_fixed_array_test.v
@@ -1,0 +1,13 @@
+type Mat4 = [16]f32
+
+struct GameObject {
+mut:
+	transform Mat4
+	children  []&GameObject
+}
+
+fn test_array_map_to_fixed_array() {
+	mut v1 := &GameObject{}
+	eprintln('children: ${v1.children.map(it.transform)}')
+	assert '${v1.children.map(it.transform)}' == '[]'
+}

